src/types.jl

```diff
@@ -283,23 +283,15 @@ end
 # Equality
 ==(a::ZonedDateTime, b::ZonedDateTime) = a.utc_datetime == b.utc_datetime
 isless(a::ZonedDateTime, b::ZonedDateTime) = isless(a.utc_datetime, b.utc_datetime)
+isequal(a::ZonedDateTime, b::ZonedDateTime) = isequal(a.utc_datetime, b.utc_datetime)
 
-# Note: `hash` and `isequal` assume that the "zone" of a ZonedDateTime is not being set
-# incorrectly.
-
-function hash(zdt::ZonedDateTime, h::UInt)
-    h = hash(zdt.utc_datetime, h)
-    h = hash(zdt.timezone, h)
-    return h
-end
+"""
+    hash(::ZonedDateTime, h)
 
-function isequal(a::ZonedDateTime, b::ZonedDateTime)
-    return (
-        isequal(a.utc_datetime, b.utc_datetime) &&
-        isequal(a.timezone, b.timezone) &&
-        isequal(a.zone, b.zone)
-    )
-end
+Compute an integer hash code for a ZonedDateTime by hashing the `utc_datetime` field.
+`hash(:utc_datetime, h)` is used to avoid collisions with `DateTime` hashes.
+"""
+hash(zdt::ZonedDateTime, h::UInt) = hash(zdt.utc_datetime, hash(:utc_datetime, h))
```

Comment:
Can we make this a multiline function? Also, I think `:utc_instant` is a better name